### ISSUE TITLE
[Bug]: Fast Refresh had to perform a full reload.

### ISSUE THREAD

**Issue by timohausmann** [serwist/serwist#48]
### Provide environment information

Using next.js 14 and pages router, running `npm run dev`, following the docs, I get endless loops of refreshs. It says in the node console:
``` 
⚠ Fast Refresh had to perform a full reload. Read more: https://nextjs.org/docs/messages/fast-refresh-reload
``` 

It does not happen on the first request, but when I hit CTRL+R once.
Any idea what could be the issue / why it's refreshing the page endlessly?

### Which project is this issue for?

@serwist/next

### Link to reproduction - Issues with a link to complete (but minimal) reproduction code help us address them faster

on demand

### To reproduce

(I just upgraded from next 13 to 14 to use this lib, I hope that is not the issue, everything else seem to work normally)

* Follow this guide https://serwist.pages.dev/docs/next/getting-started
* `npm run dev`
* Open the page: everything looks good
* Hit Ctrl+R
* The site now refreshes endlessly
* To make it calm down again, I can unregister the Service Worker

### Describe the bug

Endless loop of browser reload and the warning (see above)

### Expected behavior

No endless refresh loop.

**Comment by timohausmann** [comment]
It seems to only occur in dev, so maybe I could disable the SW in dev if I'm not working on it.

**Comment by timohausmann** [comment]
Since Next' preloading etc. behaves differently in dev mode, I found it best to only test the SW in production build.
```js
const withSerwist = require("@serwist/next").default({
    swSrc: "./sw.ts",
    swDest: "public/sw.js",
    disable: process.env.NODE_ENV === 'development',
});
```

That solves the issue for me, feel free to close.

**Comment by Xardasus** [comment]
I get the same issue now, also added disable: process.env.NODE_ENV === 'development',
running dev on Next.js 14.1 "dev": "next dev --experimental-https".
Any idea for a workaround?

**Comment by DuCanhGH** [comment]
@Xardasus if you use `installSerwist` + `runtimeCaching` like in basically any Next.js example in the docs, please disable `runtimeCaching` in development mode:

```ts
runtimeCaching: process.env.NODE_ENV === "development" ? undefined : defaultCache,
```

This feature can cause JS assets to be accidentally outdated, which is not something you want in development mode.

You may also clear the caches just to be in case.

In Serwist v9, `runtimeCaching` will be disabled by default in development mode, but this won't be backported to Serwist v8.

Hope this helps!

**Comment by Xardasus** [comment]
Many thanks @DuCanhGH, looks like it's working

**Comment by Alipourzadi** [comment]
> @Xardasus if you use `installSerwist` + `runtimeCaching` like in basically any Next.js example in the docs, please disable `runtimeCaching` in development mode:
> 
> ```ts
> runtimeCaching: process.env.NODE_ENV === "development" ? undefined : defaultCache,
> ```
> 
> This feature can cause JS assets to be accidentally outdated, which is not something you want in development mode.
> 
> You may also clear the caches just in case.
> 
> In Serwist v9, `runtimeCaching` will be disabled by default in development mode, but this won't be backported to Serwist v8.
> 
> Hope this helps!

@DuCanhGH Appreciate for your time , I get the same error though i use serwist v9, I also tried to disable ```runtimeCaching``` in dev mode but it won't help!!!

**Comment by mi-hol** [comment]
> In Serwist v9, `runtimeCaching` will be disabled by default in development mode, but this won't be backported to Serwist v8.

@DuCanhGH to make the fact "only V8 affected & will not be fixed in this version" obvious to new potential users I'd suggest:
- create labels for each major version (i.e. "V8.x", "V9.x")
- add label identifying affected version(s) to each open issue (here V8.x)
- add label "won't fix" to this issue and remove labels "help wanted" and "no stale" here

If you agree, I'd be happy to update ReadMe and add a chapter explaining above in depth

**Comment by DuCanhGH** [comment]
@mi-hol fine by me :) I'll add the labels myself, perhaps.

**Comment by mi-hol** [comment]
> I'll add the labels

Great, maintaining labels requires privileges a standard user doesn't have.

> workflows that automatically add those labels when a new issue is opened

That is feasible, i.e. any new issue should get the label "triage" , that gets removed by a trusted person (member of project organization) only after it was reproduced (then leading to label "bug") or gets rejected7moved to discussion
